robots/pkg/circuitbreaker/circuitbreaker.go

```diff
@@ -0,0 +1,81 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2022 Red Hat, Inc.
+ *
+ */
+
+package circuitbreaker
+
+import (
+	"fmt"
+	"sync"
+	"time"
+)
+
+// CircuitBreaker is a minimal implementation of the [circuit breaker pattern].
+//
+// [circuit breaker pattern]: https://martinfowler.com/bliki/CircuitBreaker.html
+type CircuitBreaker struct {
+	mutex      sync.Mutex
```

Comment:
How about using `sync.RWMutex`? It will allow multiple reader at the same time, but only a single writer. https://pkg.go.dev/sync#RWMutex
And use only `RLock()` in `isOpenAndNotFeasibleForRetry()`?

Reply:
TIL! Thanks!